From cornelia.huck@de.ibm.com  Tue Apr 17 09:44:49 2007
From: Cornelia Huck <cornelia.huck@de.ibm.com>
Date: Tue, 17 Apr 2007 13:01:38 +0200
Subject: qeth: Remove usage of subsys.rwsem
To: Greg K-H <greg@kroah.com>, Frank Pavlic <fpavlic@de.ibm.com>
Cc: Linux Netdev List <netdev@vger.kernel.org>
Message-ID: <20070417130138.37089b7b@gondolin.boeblingen.de.ibm.com>

the current driver tree contains the removal of subsys.rwsem.
Unfortunately, this breaks qeth. However, it should be no problem to
fix the walking of the devices for /proc/qeth:

No need to take subsys.rwsem during walking the devices,
driver_find_devices() should already suffice.

Signed-off-by: Cornelia Huck <cornelia.huck@de.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/s390/net/qeth_proc.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/s390/net/qeth_proc.c
+++ b/drivers/s390/net/qeth_proc.c
@@ -37,7 +37,6 @@ qeth_procfile_seq_start(struct seq_file 
 	struct device *dev = NULL;
 	loff_t nr = 0;
 
-	down_read(&qeth_ccwgroup_driver.driver.bus->subsys.rwsem);
 	if (*offset == 0)
 		return SEQ_START_TOKEN;
 	while (1) {
@@ -53,7 +52,6 @@ qeth_procfile_seq_start(struct seq_file 
 static void
 qeth_procfile_seq_stop(struct seq_file *s, void* it)
 {
-	up_read(&qeth_ccwgroup_driver.driver.bus->subsys.rwsem);
 }
 
 static void *
